portcontrolwidget.h

```diff
@@ -0,0 +1,38 @@
+#ifndef PORTCONTROLWIDGET_H
+#define PORTCONTROLWIDGET_H
+
+#include <QFrame>
+#include <QSet>
+#include <QSerialPortInfo>
+#include "boardwithleds.h"
+
+namespace Ui {
+class PortControlWidget;
+}
+
+class PortControlWidget : public QFrame
+{
+    Q_OBJECT
+
+public:
+    explicit PortControlWidget(QWidget *parent = nullptr);
+    ~PortControlWidget();
+
+    void updateAvailablePorts();
+    QString getPortName() const;
+    void setBoard(BoardWithLeds *board);
+
+signals:
+    void connectionStatusChanged(bool isConnected);
+    void portsAvailable(bool available);
+
+private slots:
+    void on_connectWidget_clicked();

```

Comment:
make the naming of your methods and variables consistent: `onConnectWidgetClicked`

Reply:
I get the consistency thing, but on_connectWidget_clicked is just how Qt Designer names slots. If I change them in the .h and .cpp files, it breaks the UI since Qt can't find them anymore.

Reply:
Any tips on doing that without breaking the UI?

Reply:
You can create the connection manually in the code.

Reply:
Should I rename all slots and connect them manually? Won’t that just add redundant code?

Reply:
Yes, exactly. It's a common practice and it's the only way to get rid of autogenerated names.